x-pack/plugins/cloud_security_posture/public/pages/compliance_dashboard/compliance_dashboard.test.tsx

```diff
@@ -196,27 +203,71 @@ describe('<ComplianceDashboard />', () => {
     );
   };
 
-  it('shows noDataConfig when status is not deployed', () => {
+  it('no findings state: not-deployed - shows NotDeployed instead of dashboard', () => {
```

Comment:
A lot of these tests repeat here and in the findings page, and they are testing functionality that is implemented in another component (the `NoFindingsStates` component). I suggest extracting the tests that test `NoFindingsStates` functionality into tests for that component. The only test that should stay here and in the findings tests is if the `NoFindingsStates` is rendered when `status !== 'indexed'` and if the page itself is rendered when `status == 'indexed'`

Reply:
there's a reason for that, those pages need to display those states while benchmarks and rules don't. those tests do share the same logic, but they also test the placement on the relevant pages